import-service/firewall.tf

```diff
@@ -0,0 +1,105 @@
+  action       = "ALLOW"
+  description  = "Broad office ips"
+  source_range = element(var.broad_range_cidrs, count.index)
+}
+
+# import-service must whitelist back-rawls in each environment
+
+data "google_compute_instance" "back_rawls" {
+  name    = var.back_rawls_instance
+  project = "broad-dsde-${var.env}"
+  zone    = "us-central1-a"
+}
+
+resource "google_app_engine_firewall_rule" "back_rawls_firewall" {
+  project      = google_app_engine_application.gae_import_service.project
+  priority     = 1000 + length(var.broad_range_cidrs)
+  action       = "ALLOW"
+  description  = "back-rawls vm"
+  source_range = "${data.google_compute_instance.back_rawls.network_interface.0.access_config.0.nat_ip}"
+}
+
+# import-service needs to whitelist pubsub
+resource "google_app_engine_firewall_rule" "pubsub_firewall" {
+  project      = google_app_engine_application.gae_import_service.project
+  priority     = 1020
+  action       = "ALLOW"
+  description  = "pubsub"
+  source_range = var.pubsub_ip_range
+}
+
+# Import service must allow traffic from each firecloud orchestration instances in an environment
+
+# look up the ip of each orch instance
+data "google_compute_instance" "orchestration" {
+  count   = length(var.orchestration_instances)
+  name    = var.orchestration_instances[count.index]
+  project = "broad-dsde-${var.env}"
+  zone    = "us-central1-a"
+}
+
+resource "google_app_engine_firewall_rule" "orchestration_firewall" {
+  count = length(var.orchestration_instances)
+
+  project      = google_app_engine_application.gae_import_service.project
+  priority     = 1030 + count.index
+  action       = "ALLOW"
+  description  = "firecloud-orchestration vms"
+  source_range = "${data.google_compute_instance.orchestration[count.index].network_interface.0.access_config.0.nat_ip}"
+}
+
+# Whitelist terra GKE cluster egress IPs
+
+# Load remote state for each cluster/workspace from the terra-cluster root module
+# Access like this:
+#   data.terraform_remote_state.terra-cluster[$cluster].outputs.egress_ips
+data "terraform_remote_state" "cluster" {
+  for_each = var.clusters_to_whitelist
+
+  backend   = "gcs"
+  workspace = each.key # "integration" cluster uses "integration" workspace etc
+
+  config = {
+    bucket = local.remote_state_bucket
+    prefix = "${local.remote_state_path}/terra-cluster"
+
+    credentials = var.google_credentials
+  }
+}
+
+# Extract the cluster egress ips from terraform state and transform data to a form usable by the 
+# app engine firewall resource. Takes a list of cluster names and outputs a list of egress ips
+# from each of those clusters 
+locals {
+  cluster_egress_outputs = [for cluster in data.terraform_remote_state.cluster : tolist(cluster.outputs.egress_ips)]
+  egress_ips             = flatten([for ip in local.cluster_egress_outputs : ip])
+}
+
+# Whitelisting the terra k8s cluster egress ips.
+# This is needed to communicate with back rawls once it is migrated to k8s.
```

Comment:
Import Service makes requests to Rawls (the cluster, not directly to back-rawls) and to Sam. I thiiiiink the TF code doesn't care, but maybe update the comment?

Reply:
Yes this will account for any services deployed on k8s that import service needs to communicate with. Updated the comment as you suggested though